cpp/arcticdb/column_store/chunked_buffer.hpp

```diff
@@ -481,7 +481,9 @@ class ChunkedBufferImpl {
             block_offsets_.clear();
         }
         block->abandon();
-        delete block;
+        auto timestamp = block->timestamp_;
+        block->~MemBlock();
+        Allocator::free(std::make_pair(reinterpret_cast<uint8_t *>(block), timestamp));
```

Comment:
```suggestion
        free_block(block);
```
and in `truncate_first_block` and `truncate_last_block` too